sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -35,3 +36,47 @@ case class Size(child: Expression) extends UnaryExpression with ExpectsInputType
     nullSafeCodeGen(ctx, ev, c => s"${ev.primitive} = ($c).size();")
   }
 }
+
+case class ArrayContains(left: Expression, right: Expression) extends BinaryExpression with ExpectsInputTypes {
+  override def dataType: DataType = BooleanType
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(ArrayType, AnyDataType)
```

Comment:
If I do exact type checking (like I was doing before, waiting for compile/assembly of the code you just posted), this fails:
```python
df = sqlContext.createDataFrame([([1, 2, 3],), ([],)], ['data'])
df.select(array_contains(df.data, 1)).collect()
```